lib/phoenix/transports/long_poller/server.ex

```diff
@@ -103,6 +108,13 @@ defmodule Phoenix.Transports.LongPoller.Server do
   end
 
   @doc """
+  Detects disconnect broadcasts and shuts down
+  """
+  def handle_info(%Broadcast{event: "disconnect"}, state) do
+    {:stop, :normal, state}
```

Comment:
:+1: Fixed. In the cowboy case, we can only return `{:shutdown, req, state}`. I assume the links are handled properly in these cases by cowboy's shutdown reason?